openssl-dynamic/src/main/c/tcn.h

```diff
@@ -103,6 +104,14 @@ jstring         tcn_new_stringn(JNIEnv *, const char *, size_t);
 
 #define AJP_TO_JSTRING(V)   (*e)->NewStringUTF((e), (V))
 
+#define TCN_CHECK_NULL(V, M, R)                      \
+    TCN_BEGIN_MACRO                                  \
+        if (V == NULL) {                             \
+            tcn_ThrowIllegalArgumentException(e, M); \
+            return R;                                \
```

Comment:
return where R is undefined is identical to simple "void" return? pretty cool trick.